src/examples/HelloSPTP.cpp

```diff
@@ -96,26 +118,55 @@ void run(UInt EPOCHS = 5000) {
     enc.encodeIntoArray(r, input.data());
     tEnc.stop();
 
-    //SP
-    tSP.start();
+    //SP (global x local) 
+    tSPloc.start();
+    fill(outSP.begin(), outSP.end(), 0);
+    spLocal.compute(input.data(), true, outSP.data());
+    spLocal.stripUnlearnedColumns(outSP.data());
+    tSPloc.stop();
+
+    tSPglob.start();
     fill(outSP.begin(), outSP.end(), 0);
-    sp.compute(input.data(), true, outSP.data());
-    sp.stripUnlearnedColumns(outSP.data());
-    tSP.stop();
+    spGlobal.compute(input.data(), true, outSP.data());
+    spGlobal.stripUnlearnedColumns(outSP.data());
+    tSPglob.stop();
+
 
-    //TP
+    //TP (TP x BackTM x TM)
     tTP.start();
     rIn = VectorHelpers::castVectorType<UInt, Real>(outSP);
     tp.compute(rIn.data(), rOut.data(), true, true);
     outTP = VectorHelpers::castVectorType<Real, UInt>(rOut);
     tTP.stop();
 
-    //Anomaly
+    tBackTM.start();
+    backTM.compute(rIn.data(), true /*learn*/, true /*infer*/);
+    const auto backAct = backTM.getActiveState();
+    const auto backPred = backTM.getPredictedState();
+    const vector<char> vAct(backAct, backAct + backTM.getNumCells());
+    const vector<char> bPred(backPred, backPred + backTM.getNumCells());
+    tBackTM.stop();
+
+    tTM.start();
+    tm.compute(COLS, outSP.data(), true /*learn*/);
+    const auto tmAct = tm.getActiveCells();
+    const auto tmPred = tm.getPredictedCells();
+    //TODO assert tmAct == spOut
+    //TODO merge Act + Pred and use for anomaly from TM
+    tTM.stop();
+ 
+
+    //Anomaly (pure x likelihood)
     tAn.start();
     res = an.compute(outSP /*active*/, prevPred_ /*prev predicted*/);
```

Comment:
for anomaly computation, I should use "current active columns, and predicted from T-1". 

1) I should convert cells back to columns, right? 
2) should I use T_i = SP.outputActiveColumns. And for T_i-1? tmPred from past round, or tmPred + tmActive from past round?